From daea5bc265829509ae9e47b0c9d2388616bfa710 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C4=81rti=C5=86=C5=A1?= <macs.martins@gmail.com>
Date: Mon, 25 Jul 2016 00:25:31 +0300
Subject: [PATCH 1/2] If container's size becomes too small while resizing in
 tiling mode (using commands), set it to the minimum size (instead of aborting
 resize operation).

---
 src/commands.c | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/src/commands.c b/src/commands.c
index 0faf277..a172456 100644
--- a/src/commands.c
+++ b/src/commands.c
@@ -546,15 +546,20 @@ static bool cmd_resize_tiling_direction(I3_CMD, Con *current, const char *way, c
     LOG("new_second_percent = %f\n", new_second_percent);
     /* Ensure that the new percentages are positive and greater than
      * 0.05 to have a reasonable minimum size. */
-    if (definitelyGreaterThan(new_first_percent, 0.05, DBL_EPSILON) &&
-        definitelyGreaterThan(new_second_percent, 0.05, DBL_EPSILON)) {
-        first->percent += ((double)ppt / 100.0);
-        second->percent -= ((double)ppt / 100.0);
-        LOG("first->percent after = %f\n", first->percent);
-        LOG("second->percent after = %f\n", second->percent);
-    } else {
-        LOG("Not resizing, already at minimum size\n");
-    }
+    if (!definitelyGreaterThan(new_first_percent, 0.05, DBL_EPSILON)) {
+        new_second_percent -= 0.05 - new_first_percent;
+        new_first_percent = 0.05;
+        LOG("Fixing up percentages because new_first_percent < 0.05\n");
+    } else if (!definitelyGreaterThan(new_second_percent, 0.05, DBL_EPSILON)) {
+        new_first_percent -= 0.05 - new_second_percent;
+        new_second_percent = 0.05;
+        LOG("Fixing up percentages because new_second_percent < 0.05\n");
+    }
+
+    first->percent = new_first_percent;
+    second->percent = new_second_percent;
+    LOG("first->percent after = %f\n", first->percent);
+    LOG("second->percent after = %f\n", second->percent);
 
     return true;
 }
-- 
2.7.3

